src/app/pages/send-tokens/hooks/use-selected-asset.ts

```diff
@@ -1,26 +1,14 @@
-import { useCallback, useMemo } from 'react';
+import { useMemo } from 'react';
 import BigNumber from 'bignumber.js';
 
-import { AssetWithMeta } from '@app/common/asset-types';
-import { getFullyQualifiedAssetName, getTicker, initBigNumber } from '@app/common/utils';
+import { getTicker, initBigNumber } from '@app/common/utils';
 import { ftDecimals, stacksValue } from '@app/common/stacks-utils';
 import { useCurrentAccountAvailableStxBalance } from '@app/query/balance/balance.hooks';
-import { useSelectedAssetItem, useUpdateSelectedAsset } from '@app/store/assets/asset.hooks';
-import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { useSelectedAssetMetadata } from '@app/store/assets/asset.hooks';
 
-export function useSelectedAsset() {
-  const selectedAsset = useSelectedAssetItem();
-  const setSelectedAsset = useUpdateSelectedAsset();
+export function useSelectedAsset(assetId?: string) {
```

Comment:
Is there anyway we could invoke this hook conditionally (within a diff component), so we could avoid having to make this an optional argument?

Reply:
Can you clarify bc sounds like you are asking me to invoke a hook conditionally which you can't do, so not sure I'm understanding?

Reply:
So you can't call a hook conditionally, but you can conditionally render a component with the hook in it, only when the data is defined, avoiding the need to handle the case when there's no `assetId`.

If this is possible to structure the components logically in this way, then that'd be better than building the hook to always handle the `undefined` case. If it makes things particularly complicated, this is okay, but worth considering.

Reply:
This actually doesn't need to be undefined so removing.